### PR TITLE
Prepare for re-usable contextmenu/dropdown items

### PR DESCRIPTION
**Pull request by jelly**
Prepare for the contextmenu and the dropdown menu to be merged together into one component. For this to happen drop everything not required from the sidebar menu.

After that we need to fix the insanity which is the current selected state. Right clicking sets the current path into state while the sidebar just assumes it acts on the current dir. Once that is refactored we can combine the actions into one and drop a ton of duplicate code.

New UI:

image

### CODE REVIEW

File: src/navigatorBreadcrumbs.jsx [hunk 1]
```diff
@@ -17,12 +17,84 @@
...
+const _ = cockpit.gettext;
+
+const SettingsDropdown = ({ showHidden, setShowHidden }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const onToggleClick = () => setIsOpen(!isOpen);
+    const onSelect = (_event, _itemId) => setIsOpen(false);
+
+    const onToggleHidden = () => {
+        setShowHidden(showHidden => {
+            localStorage.setItem("cockpit-navigator.showHiddenFiles", !showHidden ? "true" : "false");
```
Comment (cockpituous) [inline src/navigatorBreadcrumbs.jsx:44]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-254-20240207-153544-128ec524-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 2]
```diff
@@ -88,11 +88,7 @@ const ContextMenuItems = ({ path, currentDir, selected, setSelected, setHistory,
 
     const _createDirectory = () => createDirectory(Dialogs, currentDir);
     const _createLink = () => createLink(Dialogs, currentDir, files, selected[0]);
-    const _copyItem = () => {
-        copyItem(setClipboard, selected.length > 1
-            ? selected.map(s => currentDir + s.name)
-            : [currentDir + selected[0].name]);
-    };
+    const _copyItem = () => copyItem(setClipboard, selected.map(s => currentDir + s.name));
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:91]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-254-20240207-165317-8e15a54a-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigatorBreadcrumbs.jsx (hunk 1 above)
Comment (cockpituous) [inline src/navigatorBreadcrumbs.jsx:44]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-254-20240207-165317-8e15a54a-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by garrett** [comment]
I'm not sure if this applies to the PR or should be in a follow-up, but it's related to the context menus on the page:

1. Right clicking anywhere on the page when a context menu is being displayed should dismiss the menu instead of opening the browser context menu (which is what happens when you right click on the properties area or anywhere not within the grid view)
2. Opening a menu should dismiss other menus.

image

**Comment by jelly** [comment]
> I'm not sure if this applies to the PR or should be in a follow-up, but it's related to the context menus on the 

The contextmenu now comes from Cockpit which treats a `click` as:
```
        const _handleClick = (event) => {
            if (event && event.button === 0) {
                const wasOutside = !(event.target.contains === root.current);

                if (wasOutside)
                    setVisible(false);
            }
        };
```

So only the left button is respected

>     1. Right clicking anywhere on the page when a context menu is being displayed should dismiss the menu instead of opening the browser context menu (which is what happens when you right click on the properties area or anywhere not within the grid view)

Agreed, needs a fix in Cockpit (also affects terminal page)

>     2. Opening a menu should dismiss other menus.
>
> image

Agreed, also needs a fix in Cockpit.

File: src/sidebar.jsx [hunk 3]
```diff
@@ -244,15 +244,6 @@ const DropdownWithKebab = ({
         ...selected.length === 0
             ? [{ id: "show-hidden-items", onClick: onToggleHidden, title: showHiddenItems }]
             : [],
-        {
-            id: "copy-path",
-            onClick: () => {
-                navigator.clipboard.writeText(path.join("/") + "/" + (selected.length === 1
```
Comment (martinpitt) [inline src/sidebar.jsx:250]: No veto, but the commit message is confusing. It says "*should be part of*", but is it actually? This should be definite, either "is", or "will be" if you are going to add it back later on.

File: src/fileActions.jsx [hunk 4]
```diff
@@ -98,8 +98,8 @@ export const editPermissions = (Dialogs, options) => {
     );
 };
 
-export const copyItem = (setClipboard, sourcePath) => {
-    setClipboard(sourcePath);
+export const copyItem = (setClipboard, sourcePaths) => {
```
Comment (martinpitt) [inline src/fileActions.jsx:101]: At that level of pickiness it should also be "copyItems" :grin:

**Review by martinpitt — APPROVED** [review]
Thanks! This only reviews the "how", I trust you on the "why". Two nitpicks, feel free to update (and get a quick reapproval) or land as-is.

**Review by garrett — APPROVED** [review]
Aside from the issues I pointed out above (which are now in a separate issue), I think this is fine. Thanks!

File: src/navigator-card-body.jsx [hunk 5]
```diff
@@ -88,11 +88,7 @@ const ContextMenuItems = ({ path, currentDir, selected, setSelected, setHistory,
 
     const _createDirectory = () => createDirectory(Dialogs, currentDir);
     const _createLink = () => createLink(Dialogs, currentDir, files, selected[0]);
-    const _copyItem = () => {
-        copyItem(setClipboard, selected.length > 1
-            ? selected.map(s => currentDir + s.name)
-            : [currentDir + selected[0].name]);
-    };
+    const _copyItems = () => copyItems(setClipboard, selected.map(s => currentDir + s.name));
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:91]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-254-20240212-112317-82181124-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigatorBreadcrumbs.jsx (hunk 1 above)
Comment (cockpituous) [inline src/navigatorBreadcrumbs.jsx:44]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-254-20240212-112317-82181124-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/sidebar.jsx [hunk 6]
```diff
@@ -352,7 +313,7 @@ const DropdownWithKebab = ({
     const multiDropdownOptions = [
         {
             id: "copy-item",
-            onClick: () => copyItem(setClipboard, selected.map(s => path.join("/") + "/" + s.name)),
+            onClick: () => copyItems(setClipboard, selected.map(s => path.join("/") + "/" + s.name)),
```
Comment (cockpituous) [inline src/sidebar.jsx:316]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-254-20240212-112317-82181124-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**